packages/next-typesafe-url/src/index.ts

```diff
@@ -1,15 +1,32 @@
 // !!! huge credit to yesmeck https://github.com/yesmeck/remix-routes as well as Tanner Linsley https://tanstack.com/router/v1 for the inspiration for this
-
 import { generateParamStringFromObject, fillPath } from "./utils";
 import type {
   AllRoutes,
   PathOptions,
-  AppRouter as AR,
-  DynamicRoute as DR,
-} from "./types";
+  AppRouter,
+  InferRoute,
+  DynamicRoute,
+  InferPagePropsType,
+  InferLayoutPropsType,
+  UseParamsResult,
+  UseAppParamsResult,
+  ServerParseParamsResult,
+  DynamicLayout,
+} from "./static-types";
 
-export type AppRouter = AR;
-export type DynamicRoute = DR;
+export type {
+  AllRoutes,
+  PathOptions,
+  AppRouter,
+  InferRoute,
+  DynamicRoute,
+  InferPagePropsType,
+  InferLayoutPropsType,
+  UseParamsResult,
+  UseAppParamsResult,
+  ServerParseParamsResult,
+  DynamicLayout,
```

Comment:
I hope I understand it correctly.
As I user of this library, I expect to import types from the root of "next-typesafe-url", not "next-typesafe-url/app".
I don't want to import internal types as you mention. But I don't know what types are internal in this library - so it would be helpful if you tell me what you think - which type to export (maybe we need to have more examples).

> Original my intention was to keep all of the types/functions only ever used in 'app mode' in /app and vice verse with pages, with shared stuff in index. (ie the Infer...PropsType would live in /app, while AppRouter is in /index)

In the long run, I don't see why would we use the app/pages flags at all. These are just implementation details. As next.js does not allow us to declare a page lives in the pages directory and have representation in the app directory - so I think that concept should not exist at all.

I don't want to import app/pages specific types if there is a type that handle both cases. Which types should I export?